snapcraft/_store.py

```diff
@@ -59,19 +112,23 @@ def _login(store, acls=None, save=True):
         try:
             store.login(email, password, acls=acls, save=save)
             print()
-            logger.info(
-                'We strongly recommend enabling multi-factor authentication: '
-                'https://help.ubuntu.com/community/SSO/FAQs/2FA')
+            logger.info(storeapi.constants.TWO_FACTOR_WARNING)
         except storeapi.errors.StoreTwoFactorAuthenticationRequired:
             one_time_password = input('Second-factor auth: ')
             store.login(
                 email, password, one_time_password=one_time_password,
                 acls=acls, save=save)
-    except (storeapi.errors.InvalidCredentialsError,
-            storeapi.errors.StoreAuthenticationError):
-        print()
-        logger.info('Login failed.')
-        return False
+
+        # Continue if agreement and namespace conditions are met.
+        if not _check_dev_agreement_and_namespace_statuses(store):
```

Comment:
just deal with the exception in the exception catching logic a couple lines below